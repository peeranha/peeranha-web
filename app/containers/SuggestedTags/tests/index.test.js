import { SuggestedTags } from '../index';

const cmp = new SuggestedTags();

cmp.props = {
  locale: 'en',
  match: {
    params: {
      communityid: 1,
    },
  },
  tags: [],
  tagsLoading: false,
  getSuggestedTagsDispatch: jest.fn(),
  upVoteDispatch: jest.fn(),
  downVoteDispatch: jest.fn(),
};

const ev = {
  currentTarget: {
    dataset: {
      communityid: null,
    },
    id: null,
  },
};

describe('<SuggestedTags />', () => {
  it('componentDidMount', () => {
    cmp.componentDidMount();
    expect(cmp.props.getSuggestedTagsDispatch).toHaveBeenCalledWith(
      cmp.props.match.params.communityid,
    );
  });

  it('downVote', () => {
    const { communityid } = cmp.props.match.params;

    const id = 'id';
    const tagid = 'tagid';

    ev.currentTarget.dataset.tagid = tagid;
    ev.currentTarget.id = id;

    cmp.downVote(ev);
    expect(cmp.props.downVoteDispatch).toHaveBeenCalledWith(
      communityid,
      tagid,
      id,
    );
  });

  it('upVote', () => {
    const { communityid } = cmp.props.match.params;

    const tagid = 'tagid';
    const id = 'id';

    ev.currentTarget.dataset.tagid = tagid;
    ev.currentTarget.id = id;

    cmp.upVote(ev);
    expect(cmp.props.upVoteDispatch).toHaveBeenCalledWith(
      communityid,
      tagid,
      id,
    );
  });

  describe('render', () => {
    it('tagsLoading TRUE', () => {
      cmp.props.tagsLoading = true;
      expect(cmp.render()).toMatchSnapshot();
    });

    it('communitiesLoading FALSE | tags.length', () => {
      cmp.props.tagsLoading = false;
      cmp.props.tags = [{}];
      expect(cmp.render()).toMatchSnapshot();
    });

    it('communitiesLoading FALSE | !tags.length', () => {
      cmp.props.tagsLoading = false;
      cmp.props.tags = [];
      expect(cmp.render()).toMatchSnapshot();
    });
  });
});
