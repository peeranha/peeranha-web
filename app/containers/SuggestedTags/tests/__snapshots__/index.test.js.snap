// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuggestedTags /> render communitiesLoading FALSE | !tags.length 1`] = `
<div
  className="container"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Suggested tags
    </title>
    <meta
      content="Suggested tags | Description"
      name="description"
    />
  </HelmetWrapper>
  <NoSuggestedTags />
  <SuggestedTagsView
    downVote={[Function]}
    tags={Array []}
    upVote={[Function]}
  />
</div>
`;

exports[`<SuggestedTags /> render communitiesLoading FALSE | tags.length 1`] = `
<div
  className="container"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Suggested tags
    </title>
    <meta
      content="Suggested tags | Description"
      name="description"
    />
  </HelmetWrapper>
  <SuggestedTagsView
    downVote={[Function]}
    tags={
      Array [
        Object {},
      ]
    }
    upVote={[Function]}
  />
</div>
`;

exports[`<SuggestedTags /> render tagsLoading TRUE 1`] = `
<div
  className="container"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Suggested tags
    </title>
    <meta
      content="Suggested tags | Description"
      name="description"
    />
  </HelmetWrapper>
  <LoadingIndicator />
</div>
`;
